probe-rs/src/architecture/arm/core/registers/aarch32.rs

```diff
@@ -0,0 +1,302 @@
+use once_cell::sync::Lazy;
+
+use crate::{
+    core::{RegisterDataType, RegisterRole},
+    CoreRegister, RegisterFile, RegisterId,
+};
+
+use super::cortex_m::ARM32_COMMON_REGS_VEC;
+
+pub(crate) static AARCH32_COMMON_REGS: Lazy<RegisterFile> = Lazy::new(|| {
+    RegisterFile::new(
+        (*ARM32_COMMON_REGS_VEC)
+            .iter()
+            .chain(&(*AARCH32_COMMON_REGS_VEC))
+            .cloned()
+            .collect(),
+    )
+});
+
+pub(crate) static AARCH32_FP_16_REGS: Lazy<RegisterFile> = Lazy::new(|| {
+    RegisterFile::new(
+        (*ARM32_COMMON_REGS_VEC)
+            .iter()
+            .chain(&(*AARCH32_COMMON_REGS_VEC))
+            .cloned()
+            .collect::<Vec<_>>()
+            .iter()
+            .chain(&(*AARCH32_FP_16_REGS_VEC))
+            .cloned()
+            .collect(),
+    )
+});
+
+pub(crate) static AARCH32_FP_32_REGS: Lazy<RegisterFile> = Lazy::new(|| {
+    RegisterFile::new(
+        (*ARM32_COMMON_REGS_VEC)
+            .iter()
+            .chain(&(*AARCH32_COMMON_REGS_VEC))
+            .cloned()
+            .collect::<Vec<_>>()
+            .iter()
+            .chain(&(*AARCH32_FP_16_REGS_VEC))
+            .cloned()
+            .collect::<Vec<_>>()
+            .iter()
+            .chain(&(*AARCH32_FP_32_REGS_VEC))
+            .cloned()
+            .collect(),
+    )
+});
+
+static AARCH32_COMMON_REGS_VEC: Lazy<Vec<CoreRegister>> = Lazy::new(|| {
+    vec![CoreRegister {
+        name: "CPSR",
+        role: Some(RegisterRole::ProcessorStatus),
+        id: RegisterId(0b1_0000),
+        data_type: RegisterDataType::UnsignedInteger,
+        size_in_bits: 32,
+    }]
+});
+
+static AARCH32_FP_16_REGS_VEC: Lazy<Vec<CoreRegister>> = Lazy::new(|| {
+    vec![
```

Comment:
What's the reason for making this a Vec? I would prefer to keep it as a normal static array, and avoid the use of `Lazy`.

Reply:
Done :)